src/main/java/io/github/anthorx/parquet/sql/api/SQLParquetReader.java

```diff
@@ -100,19 +100,20 @@ public Record read() throws IOException {
     Record result = currentParquetReader.read();
 
     if (result == null) {
-      return readFromNextParquetReader();
+      result = readFromNextParquetReader();
     }
 
     return result;
   }
 
   private Record readFromNextParquetReader() throws IOException {
+    Record result = null;
     if (parquetReaderIterator.hasNext()) {
       currentParquetReader = parquetReaderIterator.next();
-      return currentParquetReader.read();
-    } else {
-      return null;
+      result = read();
```

Comment:
I think it's better for simplicity and comprehension if you move the recursive call inside the read method.

Reply:
It's now better (with only one method)